shell/models/fleet.cattle.io.gitrepo.js

```diff
@@ -20,6 +20,39 @@ function quacksLikeAHash(str) {
   return false;
 }
 
+function clusterIdFromLabels(labels) {
+  const clusterNamespace = labels?.[FLEET_ANNOTATIONS.CLUSTER_NAMESPACE];
+  const clusterName = labels?.[FLEET_ANNOTATIONS.CLUSTER];
+
+  return `${ clusterNamespace }/${ clusterName }`;
+}
+
+// bundleDeploymentResources extracts the list of resources deployed by a BundleDeployment
+function bundleDeploymentResources(status) {
+  // Use a map to avoid `find` over and over again
+  const resourceKey = (r) => `${ r.kind }/${ r.namespace }/${ r.name }`;
+
+  // status.resources includes of resources that were deployed by Fleet *and still exist in the cluster*
+  const resources = new Map((status?.resources || []).map((r) => [
```

Comment:
we would need to confirm that this doesn't break reactivity. For instance if we get a websocket update to status.resources is this function re-called? Pretty sure it should, but is a good general test case